### PR TITLE
Fix Clippy lints

### PR DESCRIPTION
**Pull request by tigercat2000**
This resolves every single error raised by [Clippy](https://github.com/rust-lang/rust-clippy) for every active crate. See commits for details, it's not really possible to list the ~240 odd issues Clippy saw- most are formatting or redundant code, there's only a few things that are functionally better.

### CODE REVIEW

File: crates/dm-langserver/src/debugger/mod.rs [hunk 1]
```diff
@@ -1089,7 +1093,7 @@ handle_request! {
             DebugClient::Extools(extools) => {
                 let extools = extools.get()?;
 
-                if params.variablesReference >= 0x01_000000 {
+                if params.variablesReference >= 0x0100_0000 {
```
Comment (willox) [inline crates/dm-langserver/src/debugger/mod.rs:1096]: I'm not particularly fussed here but the strange _ from before would likely have been separating the byond type id from the object identifier.

1 byte for the type, 3 bytes for the rest

Comment (tigercat2000) [reply crates/dm-langserver/src/debugger/mod.rs:1096]: i was wondering if that was the case but the fact that there's no comments anywhere in sight indicating that was a pretty big 🤷🏻

**Comment by SpaceManiac** [comment]
I should probably fix this separately but I noticed the Hash/Eq impls for Prefab and Pop only take the keys into account and not the values.

File: crates/dmdoc/src/main.rs [hunk 2]
```diff
@@ -919,10 +919,7 @@ fn module_entry<'a, 'b>(modules: &'a mut BTreeMap<String, Module1<'b>>, path: &P
     modules.entry(module_path(path)).or_insert_with(|| Module1 {
         htmlname: module_path(path),
         orig_filename: path.display().to_string().replace('\\', "/"),
-        name: Default::default(),
-        teaser: Default::default(),
-        items_wip: Default::default(),
-        defines: Default::default(),
+        .. Default::default()
```
Comment (tigercat2000) [inline crates/dmdoc/src/main.rs:922]: Woah, I didn't know you could do this, this is neat.

File: crates/dm-langserver/src/debugger/extools.rs [hunk 3]
```diff
@@ -339,9 +339,8 @@ impl Drop for Extools {
 }
 
 fn parse_lineno(comment: &str) -> Option<i64> {
-    let prefix = "Line number: ";
```
Comment (tigercat2000) [inline crates/dm-langserver/src/debugger/extools.rs:340]: lmao I guess I wasn't paying much attention here

**Comment by tigercat2000** [comment]
👍🏻

**Pull request merged**